### PR TITLE
Ensure juju-version annotations and labels are correctly updated for …

### PR DESCRIPTION
**Pull request by ycliuhw**
## Description of change

Fix upgrade-model for caas model;

- set labels and annotations correctly and consistently;

- ensures selector and labels matching (in 2.5.7, they were not matching.)

## QA steps

- prepare a caas model in 2.5.7;
- upgrade to the model to 2.6.4;

## Documentation changes

No

## Bug reference

https://bugs.launchpad.net/juju/+bug/1830128

### CODE REVIEW

File: worker/caasoperator/caasoperator.go [hunk 1]
```diff
@@ -276,10 +276,7 @@ func (op *caasOperator) init() (*LocalState, error) {
 		if err == jworker.ErrTerminateAgent {
 			return nil, err
 		}
-		return nil, errors.Annotatef(err,
-			"failed to initialize caasoperator for %q",
-			op.config.Application,
-		)
+		return nil, errors.Trace(err)
```
Comment (wallyworld) [inline worker/caasoperator/caasoperator.go]: can we keep the annotation?

File: caas/kubernetes/provider/k8s.go [hunk 2]
```diff
@@ -1323,6 +1323,23 @@ func (k *kubernetesClient) Upgrade(appName string, vers version.Number) error {
 		c.Image = podcfg.RebuildOldOperatorImagePath(c.Image, vers)
 		existingStatefulSet.Spec.Template.Spec.Containers[i] = c
 	}
+	logger.Criticalf("existingStatefulSet.Spec.Selector.MatchLabels -> %+v", existingStatefulSet.Spec.Selector)
```
Comment (wallyworld) [inline caas/kubernetes/provider/k8s.go]: remove me

**Comment by wallyworld** [comment]
[upgrading k8s model from 2.5.7 to 2.6.2 fails](https://trello.com/c/CpYZW6tg/772-upgrading-k8s-model-from-257-to-262-fails)

**Comment by ycliuhw** [comment]
[on upgrade, update juju-version annotations](https://trello.com/c/f7JCurfa/800-on-upgrade-update-juju-version-annotations)

File: worker/caasoperator/caasoperator.go (hunk 1 above)
Comment (ycliuhw) [reply worker/caasoperator/caasoperator.go]: yeah, just added Cause in the IsNotFound check.

**Comment by ycliuhw** [comment]
http://ci.jujucharms.com/job/make-check-juju/5734/consoleFull#-23270029416634ad4-80e2-4403-a89b-6238558073ce
can't reproduce on local, weird, !!build!!

**Comment by ycliuhw** [comment]
http://ci.jujucharms.com/job/make-check-juju/5741/consoleFull#203736466016634ad4-80e2-4403-a89b-6238558073ce
controller connection timeout, !!build!!

File: worker/caasoperator/caasoperator.go [hunk 3]
```diff
@@ -286,7 +286,7 @@ func (op *caasOperator) init() (*LocalState, error) {
 
 func (op *caasOperator) loop() (err error) {
 	defer func() {
-		if errors.IsNotFound(err) {
+		if errors.IsNotFound(errors.Cause(err)) {
```
Comment (wallyworld) [inline worker/caasoperator/caasoperator.go]: don't need this - IsNotFound uses Cause() internally

Comment (ycliuhw) [reply worker/caasoperator/caasoperator.go]: yeah, removed

**Comment by ycliuhw** [comment]
$$merge$$

**Comment by ycliuhw** [comment]
Again 11:03:26 
FAIL: worker_test.go:308: workerEnvironSuite.TestCharmProfilingInfoError

$$merge$$

**Pull request merged**